src/bindings/c/ov/src/ov_c_api.cpp

```diff
@@ -613,47 +618,91 @@ bool ov_model_is_dynamic(const ov_model_t* model) {
 
 std::vector<char*> split(std::string s) {
     std::vector<char*> result;
-    std::regex delimiter("[,]");
-    std::sregex_token_iterator tokens(s.cbegin(), s.cend(), delimiter, -1);
-    std::sregex_token_iterator end;
-    for (; tokens != end; ++tokens) {
-        result.push_back(str_to_char_array(*tokens));
+
+    std::regex delimiter("[^{},]+");
+    auto words_begin = std::sregex_iterator(s.begin(), s.end(), delimiter);
+    auto words_end = std::sregex_iterator();
+    for (auto iter = words_begin; iter != words_end; ++iter) {
+        result.push_back(str_to_char_array(iter->str()));
```

Comment:
If you change the return value to a vector of strings, you can just do:
```suggestion
        result.push_back(iter->str());
```

Reply:
Agree.